### PR TITLE
fs/ftruncate : set errno as EINVAL if length is negative

### PR DESCRIPTION
**Pull request by Taejun-Kwon**
It is Same as posix, change to set EINVAL as a errno

### CODE REVIEW

File: os/fs/vfs/fs_truncate.c [hunk 1]
```diff
@@ -180,8 +180,12 @@ int ftruncate(int fd, off_t length)
 	FAR struct file *filep;
 	int ret;
 
-	/* Get the file structure corresponding to the file descriptor. */
+	if (length < 0) {
+		ret = -EINVAL;
```
Comment (jeongchanKim) [inline os/fs/vfs/fs_truncate.c:184]: How about assigning the `ret` with positive errno and calling just `set_errno(ret)` in line 204?

Comment (Taejun-Kwon) [reply os/fs/vfs/fs_truncate.c:184]: This is correct, lower level of this will return OK or negative.

Comment (jeongchanKim) [reply os/fs/vfs/fs_truncate.c:184]: But this function will return the ERROR, not -ERRORNO. It is just used for setting errno.

Comment (Taejun-Kwon) [reply os/fs/vfs/fs_truncate.c:184]: he understood after I explained, hence resolve it.

**Review by sunghan-chang — APPROVED** [review]

**Pull request merged**